transaction/core/src/account_keys.rs

```diff
@@ -70,31 +92,45 @@ impl PublicAddress {
     /// # Arguments
     /// `spend_public_key` - The user's public subaddress spend key `D`,
     /// `view_public_key` - The user's public subaddress view key  `C`,
+    /// `version` - The version number of this data
     #[inline]
-    pub fn new(spend_public_key: &RistrettoPublic, view_public_key: &RistrettoPublic) -> Self {
+    pub fn new(
+        spend_public_key: &RistrettoPublic,
+        view_public_key: &RistrettoPublic,
+        version: u32,
```

Comment:
I think forcing this version field everywhere is a mistake. There should be a per-version constructor.
I'm thinking about what happens when the version changes - that likely means the underlying fields change, which means the constructor signature would change. As such, the constructor signature is already tied to a specific version.